adapter.go

```diff
@@ -75,21 +78,26 @@ func newAdapter(network string, address string, key string,
 
 // NewAdapter is the constructor for Adapter.
 func NewAdapter(network string, address string) (*Adapter, error) {
-	return newAdapter(network, address, "casbin_rules", "", "")
+	return newAdapter(network, address, "casbin_rules", "", "", nil)
 }
 
 func NewAdapterWithUser(network string, address string, username string, password string) (*Adapter, error) {
-	return newAdapter(network, address, "casbin_rules", username, password)
+	return newAdapter(network, address, "casbin_rules", username, password, nil)
 }
 
 // NewAdapterWithPassword is the constructor for Adapter.
 func NewAdapterWithPassword(network string, address string, password string) (*Adapter, error) {
-	return newAdapter(network, address, "casbin_rules", "", password)
+	return newAdapter(network, address, "casbin_rules", "", password, nil)
 }
 
 // NewAdapterWithKey is the constructor for Adapter.
 func NewAdapterWithKey(network string, address string, key string) (*Adapter, error) {
-	return newAdapter(network, address, key, "", "")
+	return newAdapter(network, address, key, "", "", nil)
+}
+
+// NewAdapterWithTLS is the constructor for Adapter.
+func NewAdapterWithTLS(network string, address string, tlsConfig *tls.Config) (*Adapter, error) {
```

Comment:
Can the tls adapter have username and password for redis?

Reply:
@ziranl16 why not add:

```go
a, _ := redisadapter.NewAdapter("tcp", "127.0.0.1:6379")
tlsConfig *tls.Config
a.SetTlsConfig(tlsConfig)
```